crypto/asn1/tasn_new.c

```diff
@@ -141,19 +141,27 @@ int asn1_item_embed_new(ASN1_VALUE **pval, const ASN1_ITEM *it, int embed)
 #endif
     return 1;
 
+ auxerr:
+    err = ASN1_R_AUX_ERROR;
+    goto memerr;
+ auxerr1:
+    err = ASN1_R_AUX_ERROR;
+    goto memerr1;
+ auxerr2:
+    err = ASN1_R_AUX_ERROR;
  memerr2:
-    ASN1_item_ex_free(pval, it);
+    for (tt = it->templates; i > 0; i--, tt++) {
+        pseqval = asn1_get_field_ptr(pval, tt);
+        asn1_template_free(pseqval, tt);
+    }
+    asn1_do_lock(pval, -1, it);
```

Comment:
This part here seems over the top.  I would need some *very* strong arguments to find code copying acceptable, so unless you have them, I'd think that you should put back the call to `ASN1_item_ex_free`...